### PR TITLE
chore(rolldown): oxc 0.36.0

### PR DESCRIPTION
**Pull request by Boshen**
CHANGELOG: https://github.com/oxc-project/oxc/pull/7227

### CODE REVIEW

File: crates/rolldown/tests/esbuild/ts/ts_export_equals/artifacts.snap
Comment (Boshen) [inline crates/rolldown/tests/esbuild/ts/ts_export_equals/artifacts.snap:1]: This requires attention. We defaulted transforming `export =` and `import foo =` to cjs.

File: crates/rolldown/tests/esbuild/ts/ts_export_equals/diff.md [hunk 1]
```diff
@@ -20,8 +20,17 @@ console.log(import_b.default);
 ### rolldown
 ```js
 
+
+//#region b.ts
+var require_b = __commonJS({ "b.ts"(exports, module) {
```
Comment (Boshen) [inline crates/rolldown/tests/esbuild/ts/ts_export_equals/diff.md:25]: This is syntax error `{ "b.ts"(exports`

Comment (hyf0) [reply crates/rolldown/tests/esbuild/ts/ts_export_equals/diff.md:25]: Seems valid for rollup/acron https://rollupjs.org/repl/?version=4.17.1&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIydmFyJTIwcmVxdWlyZV9iJTIwJTNEJTIwX19jb21tb25KUyglN0IlNUNuJTIwJTIwJTVDJTIyYi50cyU1QyUyMihleHBvcnRzJTJDJTIwbW9kdWxlKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjBmdW5jdGlvbiUyMGZvbygpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCU3RCU1Q24lMjAlMjAlMjAlMjBtb2R1bGUuZXhwb3J0cyUyMCUzRCUyMCU1QjEyMyUyQyUyMGZvbyU1RCUzQiU1Q24lMjAlMjAlN0QlNUNuJTdEKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyaWlmZSUyMiUyQyUyMm5hbWUlMjIlM0ElMjJBJTIyJTdEJTdEJTdE.

Did you mean it's a syntax error in `oxc@0.36.0` and need to be fixed?

Comment (Boshen) [reply crates/rolldown/tests/esbuild/ts/ts_export_equals/diff.md:25]: ohhhhhhh ... this is a object method with the key "b.ts" ...

**Review by hyf0 — APPROVED** [review]
LGTM @IWANABETHATGUY cc

File: crates/rolldown/src/ast_scanner/side_effect_detector/mod.rs [hunk 2]
```diff
@@ -173,13 +173,6 @@ impl<'a> SideEffectDetector<'a> {
             if key_side_effect {
               return true;
             }
-
-            let prop_init_side_effect =
```
Comment (hyf0) [inline crates/rolldown/src/ast_scanner/side_effect_detector/mod.rs:177]: Why this logic was removed?

Comment (hyf0) [reply crates/rolldown/src/ast_scanner/side_effect_detector/mod.rs:177]: Found why: https://github.com/oxc-project/oxc/pull/7220.

**Comment by IWANABETHATGUY** [comment]
> LGTM @IWANABETHATGUY cc

looks good

**Pull request merged**